### ISSUE TITLE
Restart from latest checkpoint

### ISSUE THREAD

**Issue by sheroytata** [ECP-WarpX/WarpX#3742]
Restart from checkpoint function may need to be improved to start the simulation from the checkpoint folder.. it should not be required to specify the iteration so that a simulation may be restarted automatically from the last known good checkpoint from a folder. Although if an iteration is specified it restarts from the specified iteration.

This feature would be particularly useful if the job gets preempted by the system.

**Comment by ax3l** [comment]
Agreed. Additionally, we could implement a feature that removes all but that last good checkpoint during run. This is helpful in systems where filesystem space is very limited (under-spec'd).

**Comment by ax3l** [comment]
ccing @n01r that was also interested in this feature and could help to implement.
